flink-core/src/main/java/org/apache/flink/api/common/ExecutionConfig.java

```diff
@@ -756,7 +811,27 @@ public GlobalJobParameters getGlobalJobParameters() {
      */
     public void setGlobalJobParameters(GlobalJobParameters globalJobParameters) {
         Preconditions.checkNotNull(globalJobParameters, "globalJobParameters shouldn't be null");
-        this.globalJobParameters = globalJobParameters;
+        configuration.set(
+                GLOBAL_JOB_PARAMETERS_EXECUTION_CONFIG, convertToString(globalJobParameters));
+    }
+
+    private static String convertToString(final Serializable object) {
+        try (final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                ObjectOutputStream oos = new ObjectOutputStream(baos)) {
+            oos.writeObject(object);
+            return Base64.getEncoder().encodeToString(baos.toByteArray());
+        } catch (final IOException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    private static <T extends Serializable> T convertFrom(final String objectAsString) {
+        final byte[] data = Base64.getDecoder().decode(objectAsString);
+        try (final ObjectInputStream ois = new ObjectInputStream(new ByteArrayInputStream(data))) {
+            return (T) ois.readObject();
```

Comment:
This won't work if the GlobalJobParamaters are a user-defined class. You'll need to somehow provide the EC with the usercode CL.

Reply:
Ok, good point. In that case I'm thinking about keeping those fields as they are, and not migrating them to the `configuration` for the time being. This partial migration would be still a step towards the good direction, and making for example the errors for the disabled programatic changes better. WDYT?

Reply:
I think in that case I would prefer to leave those problematic fields as they are and migrate only the non java-serialized fields.

Reply:
sounds good to me. (I also don't see a good alternative :see_no_evil: )